From 6d926bdd011e01876f5035d436fb71654c1b970f Mon Sep 17 00:00:00 2001
From: congyuyang <congyuyang@eswincomputing.com>
Date: Thu, 6 Feb 2025 16:26:16 +0800
Subject: [PATCH 402/481] WIN2030-16412:feat: Optimize dma channel request

Changelogs:
1. Resolved a kernel crash issue that occurs when a large number of tasks are added to a task

Change-Id: I1152bf1ab8f9b0547c10cdba79ddb746def0e70f
Signed-off-by: congyuyang <congyuyang@eswincomputing.com>
---
 drivers/memory/eswin/es_dma_memcp/es_dma_memcp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/memory/eswin/es_dma_memcp/es_dma_memcp.c b/drivers/memory/eswin/es_dma_memcp/es_dma_memcp.c
index 3eec1aa413f2..cc4b81e6ef15 100644
--- a/drivers/memory/eswin/es_dma_memcp/es_dma_memcp.c
+++ b/drivers/memory/eswin/es_dma_memcp/es_dma_memcp.c
@@ -169,12 +169,12 @@ int esw_memcp_alloc_dma(struct esw_cmdq_task *task)
     dma_cap_set(DMA_MEMCPY, mask);
 
     dma_ch = dma_request_channel(mask, filter, &params);
-    if (IS_ERR(dma_ch)) {
+    if (!dma_ch) {
         pr_warn("dma request channel failed, Try using any of them.\n");
         dma_ch = dma_request_channel(mask, NULL, NULL);
     }
 
-    if (IS_ERR(dma_ch)) {
+    if (!dma_ch) {
         pr_err("dma request channel failed\n");
         return -ENODEV;
     }
-- 
2.49.0

